src/Navigation/Navigation.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Mixins } from '../Utils';
+import { NavigationHeader, NavigationItem } from '.';
+
+interface NavProps {
+    to: string;
+    label?: string;
+    icon: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
+}
+
+export interface NavigationProps {
+    label?: string;
+    header?: React.ReactElement;
+    footer?: React.ReactElement;
+    pages?: {
+        [name: string]: {
+            component: React.LazyExoticComponent<React.ComponentType>;
+            navProps: NavProps;
+        };
+    };
+}
+
+export const Navigation: React.FC<NavigationProps> = ({
+    label,
+    pages = {},
+    header,
+    footer,
+    ...props
+}) => (
+    <Container {...props}>
+        <NavigationHeader label={label} />
+        {header}
+        <Items>
+            {Object.entries(pages).map(([key, { navProps }]) => (
+                <NavigationItem key={key} {...navProps}>
+                    {key}
+                </NavigationItem>
+            ))}
```

Comment:
This should be handled externally. This locks the developer into using only NavigationItems as the children. Look back at the terminal, the logic for the generation of pages should be handled externally.

Reply:
already commented on this...